src/odemis/acq/stitching/_registrar.py

```diff
@@ -14,18 +14,534 @@
+
+LEFT = 1  # direction in pos_to_left_right function
+RIGHT = -1
+
+
+class IdentityRegistrar(object):
+    """ Returns position as-is """
+
+    def __init__(self):
+        self.tile_pos = []
+        self.dep_tiles_pos = []
+
+    def addTile(self, tile, dependent_tiles=None):
+        """ 
+        Extends grid by one tile. 
+        tile (DataArray of shape YX): each image must have at least MD_POS and MD_PIXEL_SIZE metadata. 
+        They should all have the same dtype.
+        dependent_tiles (list of DataArray or None): each of the dependent tile, where their position 
+        can be considered fixed relative to the main tile. Their content and metadata are not used
+        for the computation of the final position.
+        """
+        self.tile_pos.append(tile.metadata[model.MD_POS])
+        if dependent_tiles:
+            pos = [t.metadata[model.MD_POS] for t in dependent_tiles]
+            self.dep_tiles_pos.append(pos)
+
+    def getPositions(self):
+        """ 
+        returns:
+        tile_positions (list of N tuples of 2 floats)): the adjusted position in X/Y for each tile, in the order they were added
+        dep_tile_positions (list of N tuples of tuples of 2 floats): the adjusted position for each dependent tile 
+        (in the order they were passed)
+        """
+        return self.tile_pos, self.dep_tiles_pos
+
+
+class ShiftRegistrar(object):
+    """ 
+    Locates the position of the image relative to the previous top tiles (horizontally and vertically) 
+    by using cross-correlation. The cross-correlation is done using just the part of the images which are 
+    supposed to be overlapping. In case the cross-correlation doesn't work (based on a couple of simple tests), 
+    fallback to the average shift on the same axis. 
+    """
+
+    def __init__(self):
+        # arrays to store the vertical/horizontal shift values measured for
+        # each tile
+        # initialize grid to 1x1. The size will increase as new tiles are
+        # added.
+        self.nx = 1
+        self.ny = 1
+        self.coords_ver = [[None]]
+        self.coords_hor = [[None]]
+        self.shifts = [[None]]
+        self.tiles = [[None]]
+        self.last_ver = (0, 0)  # last vertical shift measured
+        self.last_hor = (0, 0)  # last horizontal shift measured
+
+        # Initialize overlap. This will be modified after the second tile has been added.
+        # A value is needed to avoid errors when calculating the position for
+        # the first tile.
+        self.ovrlp = 0
+
+        # List of 2D indices for grid positions in order of acquisition
+        self.acqOrder = []
+
+        self.shift_tile_dep_tiles = []  # N x K list. N: number of tiles, K: number of dep_tiles.
+        # Shift between main tile and dependent tiles
+
+        self.px_size = None  # Tuple of 2 floats. (X,Y) pixel size of the current tile
+        self.size = None  # Tuple of 2 floats. (Y,X) shape of the current tile
```

Comment:
`self.size` is a tuple of _int_. Also add that the unit is in px.
You can also mention the unit of `px_size` is in m.